pkg/controller/dataimportcron-controller.go

```diff
@@ -377,6 +379,11 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 				cc.AddAnnotation(snapshot, cc.AnnSourceVolumeMode, string(*volMode))
 			}
 		}
+		// Copy labels found on dataSource to the existing snapshot in case of upgrades.
+		dataSource, err := r.getDataSource(ctx, dataImportCron)
+		if err == nil {
```

Comment:
I think we're "forgetting" to requeue in case the GET fails for a random network err